frontend/src/views/cluster/Overview/components/OverviewRightPanel/OverviewRightPanel.vue

```diff
@@ -270,34 +287,39 @@ const {
           </Tooltip>
         </div>
       </OverviewRightPanelItem>
-      <OverviewRightPanelItem
-        v-if="talosUpgradeStatus"
-        name="Talos Version"
-        @click="openClusterUpdate(Update.Talos)"
-      >
+      <OverviewRightPanelItem v-if="talosUpgradeStatus" name="Talos Version">
         <div>{{ talosVersion }}</div>
         <Tooltip
           v-if="newTalosVersionsAvailable?.length"
-          :description="`Newer Talos versions are avalable: ${newTalosVersionsAvailable.join(', ')}`"
+          :description="
+            locked
+              ? `Newer Talos versions are available: ${newTalosVersionsAvailable.join(', ')}. However, Talos updates are disabled when the cluster is locked.`
+              : `Newer Talos versions are available: ${newTalosVersionsAvailable.join(', ')}`
+          "
         >
           <TIcon
             v-if="newTalosVersionsAvailable"
             class="h-4 w-4 cursor-pointer"
             icon="upgrade-available"
+            @click="!locked ? openClusterUpdate(Update.Talos) : undefined"
           />
         </Tooltip>
       </OverviewRightPanelItem>
-      <OverviewRightPanelItem
-        v-if="kubernetesUpgradeStatus"
-        name="Kubernetes Version"
-        @click="openClusterUpdate(Update.Kubernetes)"
-      >
+      <OverviewRightPanelItem v-if="kubernetesUpgradeStatus" name="Kubernetes Version">
         <div>{{ kubernetesVersion }}</div>
         <Tooltip
           v-if="newKubernetesVersionsAvailable?.length"
-          :description="`Newer Kubernetes versions are avalable: ${newKubernetesVersionsAvailable.join(', ')}`"
+          :description="
+            locked
+              ? `Newer Kubernetes versions are available: ${newKubernetesVersionsAvailable.join(', ')}. However, Kubernetes updates are disabled when the cluster is locked.`
+              : `Newer Kubernetes versions are available: ${newKubernetesVersionsAvailable.join(', ')}`
+          "
         >
-          <TIcon class="h-4 w-4 cursor-pointer" icon="upgrade-available" />
+          <TIcon
+            class="h-4 w-4 cursor-pointer"
+            icon="upgrade-available"
+            @click="!locked ? openClusterUpdate(Update.Kubernetes) : undefined"
```

Comment:
or maybe we can alter `openClusterUpdate` func to take into account locked flag and do `return`.